apps/websockets/viewsets.py

```diff
@@ -0,0 +1,160 @@
+import logging
+
+from django.contrib.auth import get_user_model
+from django.db.models import Q
+from django.forms import model_to_dict
+from drf_yasg.utils import swagger_auto_schema, no_body
+from rest_framework import viewsets, status
+from rest_framework.decorators import action
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from apps.events.serializers import EmptySerializer
+from apps.events.serializers.events import ParticipantSerializer
+from apps.websockets.managers import ChatManager
+from apps.websockets.models import Chat, Message
+from apps.websockets.permissions.chat import ChatPermissions
+from apps.websockets.permissions.message import MessagePermissions
+from apps.websockets.serializers.chat import (
+    ChatRetrieveSerializer,
+    ChatListSerializer,
+)
+from apps.websockets.serializers.message import (
+    MessageRetrieveSerializer,
+    MessageCreateSerializer,
+)
+from apps.websockets.utils import list_chats_raw
+
+logger = logging.getLogger("websockets_app")
+
+user_model = get_user_model()
+
+
+class ChatViewSet(viewsets.ModelViewSet):
+    model = Chat
+    permission_classes = [IsAuthenticated, ChatPermissions]
+    lookup_url_kwarg = "chat_id"
+    http_method_names = ["list", "get", "post"]
+
+    def get_serializer_class(self):
+        match self.action:
+            case "retrieve":
+                return ChatRetrieveSerializer
+            case "list":
+                return ChatListSerializer
+            case "messages":
+                return MessageRetrieveSerializer
+            case "participants":
+                return ParticipantSerializer
+            case "send_message":
+                return MessageCreateSerializer
+            case _:
+                return EmptySerializer
+
+    def get_queryset(self):
+        if self.kwargs.get("chat_id"):
+            self.queryset = self.model.objects.all()
+            return self.queryset.all()
+
+        # TODO: it was easier to write a raw query, but it requires an approve
```

Comment:
I don't mind having raw sql since most of the times it's much easier to read than complex orm queries.

Reply:
What do you think about storing raws in variables? Should i create a separate file or leave them in `utils`?

Reply:
for now it can be part of utils